### PR TITLE
Core/Spells: Melee spells against sitting targets are guarenteed as critical strikes.

### LINKED ISSUE #5986: Stealthed Attacks Against Sitting/Prone Enemy Players Do Not Crit
**CURRENT BEHAVIOUR:**

See also: https://github.com/azerothcore/azerothcore-wotlk/pull/5872

The majority of attacks vs sitting players do crit now, thanks to the above ticket. However, attacks from stealth, such as a feral druid's Ravage, still do not automatically crit vs a sitting or prone target.

Combat log showing multiple Ravages from stealth failing to crit vs a sitting enemy player:
WoWScrnShot_051821_214827

**EXPECTED BLIZZLIKE BEHAVIOUR:**

Attacks from stealth should crit vs. sitting players. 

**SOURCE:**

This is not controversial.

**STEPS TO REPRODUCE THE PROBLEM:**

1. Duel another player.
2. Have them sit or lie down.
3. Go into Prowl/Stealth and attack them from behind.
4. Note the non-critical hit inflicted.

**AC HASH/COMMIT:**
 
rev. 5b9415be5a8b 2021-05-21 10:25:13 +0000

**OPERATING SYSTEM:**

Linux Mint 20.1

**MODULES:**

N/A.

**CUSTOMIZATIONS:**

N/A.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/98598883-stealthed-attacks-against-sitting-prone-enemy-players-do-not-crit?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by UltraNix**
## Changes Proposed:
-  Melee spells against sitting targets should do crit.

## Issues Addressed:

- Closes #5986.

## Tests Performed:

- Tested ingame in duel.

## How to Test the Changes:

1. Duel another player.
2. Have them sit or lie down.
3. Go into Prowl/Stealth and attack them from behind.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by pklloveyou** [comment]
Tested

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -11598,30 +11598,36 @@ float Unit::SpellTakenCritChance(Unit const* caster, SpellInfo const* spellProto
...
                         break;
+                    }
+                    break;
                 }
+            }
+
+            // 100% critical chance against sitting target
+            if (GetTypeId() == TYPEID_PLAYER && (IsSitState() || getStandState() == UNIT_STAND_STATE_SLEEP))
+            {
+                return 100.f;
```
Comment (Kitzunu) [inline src/server/game/Entities/Unit/Unit.cpp:11629]:
```suggestion
                return 100.0f;
```

File: src/server/game/Entities/Unit/Unit.cpp [hunk 2]
```diff
@@ -11598,30 +11598,36 @@ float Unit::SpellTakenCritChance(Unit const* caster, SpellInfo const* spellProto
...
+            }
             [[fallthrough]]; // TODO: Not sure whether the fallthrough was a mistake (forgetting a break) or intended. This should be double-checked.
```
Comment (Kitzunu) [inline src/server/game/Entities/Unit/Unit.cpp:11631]: Should the fallthrough be here? Maybe you can check this todo as you are rewriting it anyways =)

Comment (UltraNix) [reply src/server/game/Entities/Unit/Unit.cpp:11631]: I'm not rewriting this - only adding one condition not related to this comment.

**Review by Helias — APPROVED** [review]

File: src/server/game/Entities/Unit/Unit.cpp (hunk 2 above)
Comment (Kitzunu) [reply src/server/game/Entities/Unit/Unit.cpp:11631]: ah, the indent change made me think you rewrote the whole case =)

**Comment by FrancescoBorzi** [comment]
@Helias don't forget to update the labels

**Pull request merged**